yt_dlp/utils.py

```diff
@@ -2075,7 +2076,8 @@ def sanitize_open(filename, open_mode):
             raise
         else:
             # An exception here should be caught in the caller
-            stream = open(encodeFilename(alt_filename), open_mode)
+            stream = locked_file(encodeFilename(filename), open_mode, block=False)
+            stream = stream.__enter__()
```

Comment:
bad design, but it's not your fault. The codebase should have used context managers